src/adapters/kafka/index.ts

```diff
@@ -0,0 +1,166 @@
+import fs from 'fs'
+import { Kafka, Consumer } from 'kafkajs'
+import Adapter from '../../lib/adapter.js'
+import GleeMessage from '../../lib/message.js'
+
+
+class KafkaAdapter extends Adapter {
+  private kafka: Kafka
+  private firstConnect: boolean = true
+  name(): string {
+    return 'Kafka adapter'
+  }
+
+  async connect() {
+    const broker = this.AsyncAPIServer.url()
```

Comment:
If it's a URL, I'd instead name it as brokerUrl:

```suggestion
    const brokerUrl = this.AsyncAPIServer.url()
```